plugins/postcss-design-tokens/src/data-formats/parse-import.ts

```diff
@@ -0,0 +1,67 @@
+import valueParser from 'postcss-value-parser';
+import { Token } from './base/token';
+import { extractStyleDictionaryTokens } from './style-dictionary/style-dictionary';
+import path from 'path';
+import { promises as fsp } from 'fs';
+
+function parseImport(statement: string): { filePath: string, format: string, conditions: Array<string> } {
+	const importAST = valueParser(statement);
+
+	const result = {
+		filePath: '',
+		format: 'standard',
+		conditions: ['6b4e71e7-4787-42f7-a092-8684961895db'], // a random, but shared default condition
+	};
+
+	importAST.walk((node) => {
+		if (node.type === 'function' && node.value === 'url') {
+			result.filePath = node.nodes[0].value;
+		}
+
+		if (node.type === 'function' && node.value === 'format') {
+			result.format = node.nodes[0].value;
+		}
+
+		if (node.type === 'function' && node.value === 'when') {
+			result.conditions = node.nodes.filter((child) => {
+				return child.type === 'string';
+			}).map((child) => child.value);
+		}
+	});
+
+	if (!result.conditions.length) {
+		result.conditions = ['6b4e71e7-4787-42f7-a092-8684961895db']; // a random, but shared default condition
+	}
+
+	return result;
+}
+
+export async function tokensFromImport(buildIs: Array<string>, sourceFilePath: string, statement: string, alreadyImported: Set<string>): Promise<{ filePath: string, tokens: Map<string, Token> }|false> {
+	const { filePath, format, conditions } = parseImport(statement);
+	if (!conditions.every((condition) => buildIs.includes(condition))) {
+		return false;
+	}
+
+	const resolvedPath = path.resolve(path.dirname(sourceFilePath), filePath);
+	if (alreadyImported.has(resolvedPath)) {
+		return false;
+	}
+
+	alreadyImported.add(resolvedPath);
+
+	const fileContents = await fsp.readFile(resolvedPath, 'utf8');
```

Comment:
This could throw, should some sort of try/catch be added here that handles failure?

Reply:
The caller has try catch as it also has access to PostCSS objects and can emit PostCSS warnings :

```js
// plugins/postcss-design-tokens/src/index.ts
let importResult: { filePath: string, tokens: Map<string, Token> }|false;
try {
	importResult = await tokensFromImport(buildIs, atRule.filePath, atRule.params, importedFiles);
	if (!importResult) {
		continue;
	}
} catch (e) {
	atRule.node.warn(result, `Failed to import design tokens from "${atRule.params}" with error:\n\t` + (e as Error).message);
	continue;
}
```

Reply:
Didn't see that! Thanks for catching this!